catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTablePartitionOperations.java

```diff
@@ -0,0 +1,329 @@
+        statement.executeUpdate(dropPartitionsSql);
+        return true;
+      }
+    } catch (SQLException e) {
+      GravitinoRuntimeException exception = exceptionConverter.toGravitinoException(e);
+      if (exception instanceof NoSuchPartitionException) {
+        return false;
+      }
+      throw exception;
+    }
+  }
+
+  private Transform getPartitionInfo(Connection connection) throws SQLException {
+    String showCreateTableSql = String.format("SHOW CREATE TABLE `%s`", tableName);
+    // Note, the connection here will be reused and should not be closed.
+    try (Statement statement = connection.createStatement();
+        ResultSet result = statement.executeQuery(showCreateTableSql)) {
+      StringBuilder createTableSql = new StringBuilder();
+      while (result.next()) {
+        createTableSql.append(result.getString("Create Table"));
+      }
+      Optional<Transform> transform =
+          DorisUtils.extractPartitionInfoFromSql(createTableSql.toString());
+      return transform.orElseThrow(
+          () -> new NotPartitionedTableException("%s is not a partitioned table", tableName));
+    }
+  }
+
+  private Partition fromDorisPartition(
+      ResultSet resultSet, Transform partitionInfo, Map<String, Type> columnTypes)
+      throws SQLException {
+    String partitionName = resultSet.getString("PartitionName");
+    String partitionKey = resultSet.getString("PartitionKey");
+    String partitionValues = resultSet.getString("Range");
+    ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
+    propertiesBuilder.put("PartitionId", resultSet.getString("PartitionId"));
+    propertiesBuilder.put("VisibleVersion", resultSet.getString("VisibleVersion"));
+    propertiesBuilder.put("VisibleVersionTime", resultSet.getString("VisibleVersionTime"));
+    propertiesBuilder.put("State", resultSet.getString("State"));
+    propertiesBuilder.put("PartitionKey", partitionKey);
+    propertiesBuilder.put("DataSize", resultSet.getString("DataSize"));
+    propertiesBuilder.put("IsInMemory", resultSet.getString("IsInMemory"));
+    ImmutableMap<String, String> properties = propertiesBuilder.build();
+
+    String[] partitionKeys = partitionKey.split(", ");
+    if (partitionInfo instanceof Transforms.RangeTransform) {
+      if (partitionKeys.length != 1) {
+        throw new UnsupportedOperationException(
+            "Multi-column range partitioning in Doris is not supported yet");
+      }
+      Type partitionColumnType = columnTypes.get(partitionKeys[0]);
+      Literal<?> lower = Literals.NULL;
+      Literal<?> upper = Literals.NULL;
+      Matcher matcher = RANGE_PARTITION_PATTERN.matcher(partitionValues);
+      if (matcher.find()) {
+        String lowerValue = matcher.group(2);
+        lower = Literals.of(lowerValue, partitionColumnType);
+      }
+      if (matcher.find()) {
+        String upperValue = matcher.group(2);
+        upper = Literals.of(upperValue, partitionColumnType);
+      }
```

Comment:
Why don't you combine these two?

Reply:
There is no other suitable way here to combine these. When we call `matcher.find()`, it would start looking for the next capture group from the location of the last successful match, so typically we will always get two matches here(for upper and lower). If we use `while(matcher.find())` here, we will have to use a flag variable to distinguish between the upper and the lower, this might be more complicated.

Reply:
Shouldn't the second `matcher.find()` be moved into the first `if` block?